### ISSUE TITLE
darksky api

### ISSUE THREAD

**Issue by rat0r** [RustyBower/sopel-weather#10]
dark sky api is no longer. the new user signups are done and its posted that they will end all free api services too. i found a possible replace for API's, they allow 1,000+ free checks on their api.... https://openweathermap.org/

thanks - sorcerer

**Comment by shogerr** [comment]
Yeah, kind of a bummer. Thanks again Apple.

**Comment by rat0r** [comment]
I know it was quite the bummer when I went to use it and I couldn’t get an API for it lmao cause I love having weather bots in my channel From: shogerrSent: Thursday, May 28, 2020 7:56 PMTo: RustyBower/sopel-weatherCc: rat0r; AuthorSubject: Re: [RustyBower/sopel-weather] darksky api (#10) Yeah, kind of a bummer. Thanks again Apple.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by MacFan4000** [comment]
@RustyBower thoughts on this?

**Comment by RustyBower** [comment]
It’s on my list of things to do. Should have it done in the next day or so.

**Comment by RustyBower** [comment]
This should be fixed in https://github.com/RustyBower/sopel-weather/commit/e94f5db6da8f01d2399493c10cfe3296cd989940

**Comment by shogerr** [comment]
Hmmm, is a zipcode intended to work? If not, then we're not quite getting a good error message:

```
[2020-06-11 16:19:02,246] sopel.bot            ERROR    - Unexpected error ('error') from ekem at 2020-06-11 16:19:02.246462. Message was: .weather 97401
Traceback (most recent call last):
  File "C:\Python38\lib\site-packages\sopel\bot.py", line 606, in call
    exit_code = func(sopel, trigger)
  File "C:\Python38\lib\site-packages\sopel_modules\weather\weather.py", line 232, in weather_command
    data = get_weather(bot, trigger)
  File "C:\Python38\lib\site-packages\sopel_modules\weather\weather.py", line 202, in get_weather
    return openweathermap_weather(bot, latitude, longitude, location)
  File "C:\Python38\lib\site-packages\sopel_modules\weather\providers\weather\openweathermap.py", line 48, in openweathermap_weather
    raise Exception('Error: {}'.format(data['error']))
KeyError: 'error'
```

If it is supposed to take a zip, then there's another problem.

**Comment by RustyBower** [comment]
Zip codes work just fine, I assume from the error you're using openweathermap?

18:26:40 <RustyCloud> .weather 90210
18:26:40 <Sopel> Beverly Hills, California, US: 29°C (85°F), Clear, Humidity: 56%, Gentle breeze 3.6m/s (↗)

The error catch is something I need to fix, so I'll push a change for that in a second. What's the exact command you're running?

**Comment by shogerr** [comment]
@RustyBower - Yeah, opemweathermap. I just tried it again, and got a really funny result: 
```
<ekem> .weather 97401
<db42> District of Banská Bystrica, Region of Banská Bystrica, SK: 15°C (59°F), Clouds, Humidity: 87%, Light air 1.5m/s (↘)
```

However, 97401 is a US zipcode. So it failed once with `.weather 97401`, and then gave me an erroneous result the next.'

Thanks for the update though! It's nice that it's working on some level : )

**Comment by MacFan4000** [comment]
@RustyBower I found an issue:
19:31 <+MacFan4000> .setlocation Portland, ME, US
19:31 <ZppixBot-test> MacFan4000: I now have you at Portland, Maine, US
19:31 <+MacFan4000> .weather
19:31 <ZppixBot-test> None: 19°C (66°F), Clouds, Humidity: 82%, Light breeze 3.1m/s (↑)
19:33 <+MacFan4000> .weather Portland, ME US
19:33 <ZppixBot-test> Portland, Maine, US: 19°C (66°F), Clouds, Humidity: 82%, Light breeze 3.1m/s (↑)

It says none when .weather uses the user set location, but not when providing a location after .weather

**Comment by RustyBower** [comment]
@shogerr Unfortunately I'm at the mercy of whatever result that openweathermap's API wants to give me back (I've run into some strange edge cases during my darksky testing as well). So, unfortunately, I would recommend setting the location via (City, State, Country) where possible to reduce the number of weird edge cases

**Comment by RustyBower** [comment]
These should be fixed in master, and once the build is done, in v1.2.2

**Comment by MacFan4000** [comment]
@RustyBower thank you so much!! This works great. :)

**Comment by RustyBower** [comment]
Fantastic. Closing this out for now, if you run into any other issues, please create another issue for me!
